src/axelrod_dojo/archetypes/hmm.py

```diff
@@ -0,0 +1,227 @@
+import random
+from random import randrange
+
+import numpy as np
+from axelrod.strategies.hmm import HMMPlayer
+from axelrod import Action
+from axelrod_dojo.utils import Params
+
+C, D = Action.C, Action.D
+
+
+def copy_lists(rows):
+    new_rows = list(map(list, rows))
+    return new_rows
+
+def random_vector(size):
+    """Create a random vector of values in [0, 1] that sums to 1."""
+    vector = []
+    s = 1
+    for _ in range(size - 1):
+        r = s * random.random()
+        vector.append(r)
+        s -= r
+    vector.append(s)
+    return vector
+
+def normalize_vector(vec):
+    s = sum(vec)
+    if s == 0.0:
+        n = len(vec)
+        return [1 /n for v in vec]
+    vec = [v / s for v in vec]
+    return vec
+
+def mutate_row(row, mutation_probability):
```

Comment:
A doc string would be nice, along with a comment about keeping the values within [0, 1]